tests/auto/jobs/build.py

```diff
@@ -18,35 +21,64 @@ def run(job_obj):
+            log_name = 'expt.out'
+            we2e_script = expt_script_loc + '/end_to_end_tests.sh'
+            if os.path.exists(we2e_script):
+                logger.info('Running end to end test')
+                create_expt_commands = \
+                    [[f'./end_to_end_tests.sh {job_obj.machine} '
+                      f'{job_obj.hpc_acc} >& '
+                      f'{log_name}', expt_script_loc]]
+                job_obj.run_commands(logger, create_expt_commands)
+                logger.info('After end_to_end script')
+                if os.path.exists(expt_dirs):
+                    job_obj.comment_append('Rocoto jobs started')
+                    process_expt(job_obj, expt_dirs)
+                else:
+                    gen_log_loc = pr_repo_loc + '/regional_workflow/ush'
+                    gen_log_name = 'log.generate_FV3LAM_wflow'
+                    process_gen(job_obj, gen_log_loc, gen_log_name)
+            else:
+                job_obj.comment_append(f'Script {we2e_script} '
+                                       'does not exist in repo')
+                job_obj.comment_append('Cannot run WE2E tests')
+    else:
+        job_obj.comment_append('Build Failed')
+    job_obj.send_comment_text()
 
 
 def set_directories(job_obj):
+    """
+    Set up work directory for various hpc machines
+    """
     logger = logging.getLogger('BUILD/SET_DIRECTORIES')
     if job_obj.machine == 'hera':
         workdir = '/scratch2/BMC/zrtrr/rrfs_ci/autoci/pr'
+
     elif job_obj.machine == 'jet':
-        workdir = '/lfs4/HFIP/h-nems/emc.nemspara/autort/pr'
-        blstore = '/lfs4/HFIP/h-nems/emc.nemspara/RT/NEMSfv3gfs/'
-        rtbldir = '/lfs4/HFIP/h-nems/emc.nemspara/RT_BASELINE/'\
-                  f'emc.nemspara/FV3_RT/REGRESSION_TEST_{job_obj.compiler.upper()}'
+        workdir = '/lfs1/BMC/nrtrr/rrfs_ci/autoci/pr'
+
     elif job_obj.machine == 'gaea':
         workdir = '/lustre/f2/pdata/ncep/emc.nemspara/autort/pr'
-        blstore = '/lustre/f2/pdata/ncep_shared/emc.nemspara/RT/NEMSfv3gfs'
-        rtbldir = '/lustre/f2/scratch/emc.nemspara/FV3_RT/'\
-                  f'REGRESSION_TEST_{job_obj.compiler.upper()}'
+
     elif job_obj.machine == 'orion':
         workdir = '/work/noaa/nems/emc.nemspara/autort/pr'
-        blstore = '/work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs'
-        rtbldir = '/work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/'\
-                  f'REGRESSION_TEST_{job_obj.compiler.upper()}'
+
     elif job_obj.machine == 'cheyenne':
         workdir = '/glade/scratch/dtcufsrt/autort/tests/auto/pr'
-        blstore = '/glade/p/ral/jntp/GMTB/ufs-weather-model/RT/NEMSfv3gfs'
-        rtbldir = '/glade/scratch/dtcufsrt/FV3_RT/'\
-                  f'REGRESSION_TEST_{job_obj.compiler.upper()}'
+
     else:
-        logger.critical(f'Machine {job_obj.machine} is not supported for this job')
+        logger.critical(f'Machine {job_obj.machine} id '
+                        'not supported for this job')
```

Comment:
Is there a way to pass in a configuration that specifies the workdir rather than `if-then-else` ing the various machines?

Reply:
Please see next comment response.